### DISCUSSION TITLE
Parts and PCB

### DISCUSSION THREAD

**Discussion by ericgrand** [dzurikmiroslav/esp32-evse#52] [General]
Hi,

Very exciting project, but unless I'm mistaken, I do not see were to order the PCB, possibily the parts and also I can't find a wiring diagram...
Would it be possible to direct me to the correct url's?

Thanks in advance!

Eric

**Comment by dzurikmiroslav** [comment]
Hi, thanks.
I was designing and ordering (only pcb) from EasyEDA, here is link https://oshwlab.com/dzurik.miroslav/esp32s2-diy-evse

**Reply by ericgrand** [reply]
Thanks, but I don't see a way to order even just the PCB on the link given...

**Reply by nagyrobi** [reply]
The procedure is as follows:
- create an account on the site
- open the link above and check if you're logged in
- click on the `Open in Editor` button, let the web based PCB editor to load
- click in `Fabrication` menu, `PCB Fabrication file(Gerber)...`
- in the message box that appears click `Yes, Check DRC`
- in the window that appears click the `One click order PCB/SMT`, in the next message box click `OK`
- the jlcpcb.com site will open and a progress bar will show uploading the file (if, not, log in to jlcpcb.com with the same username and password, close the window, and click the order button in the previous page again).
- click `Save to Cart` button and go ahead with the ordering and payment.

Apparently you can order a minimum of 5 PCBs.

**Reply by nagyrobi** [reply]
PCBs arrived in great quality.
I ordered most of the parts in a local store, except:
- T1.25A
- 10uH
- IRM-20-12
- K7803-500R3
- K7805-1000R3
- MAX3485CPA
- CT (DL-CT08CL10)

which I source from Ali (these were either unavailable in desired physical size, or way too pricey locally).
I also decided to replace the Songle SRD-12VDC-SL-C relay with Finder F 36.11.9.012.

I will order the RCM14-01 later, only after the device proves functional for me. It's an expensive part... (luckily it can be optional).

**Reply by nagyrobi** [reply]
Since the min. order was for 5 PCBs, I just can't stand against finishing them all...

![ResizedImage_2025-09-21_17-57-08_1](https://github.com/user-attachments/assets/0736dc56-9b08-4a92-81fb-d300bd8b3550)

**Reply by ericgrand** [reply]
Are you willing to sell the extra cards? What would be your price shipped to France?

**Reply by nagyrobi** [reply]
Didn't consider selling... Didn't even sum up the costs of material per card...

**Reply by nagyrobi** [reply]
Done, test successful with RCM and ESPHome-based LVGL display:
![ResizedImage_2025-09-29_15-04-41_1](https://github.com/user-attachments/assets/c8dbc712-c818-4fc9-bfb2-8f076baffe32)

**Comment by nagyrobi** [comment]
Do you publish the PCB / schematic / BOM for ESP32-S3 version?

**Reply by dzurikmiroslav** [reply]
I am designed only ESP32-S2 version, here is link to EasyEDA https://oshwlab.com/dzurik.miroslav/esp32s2-diy-evse
From EasyEDA I ordered only PCB, and board soldered at home.
Yes, ESP32-S3 would be better, but it didn't exist at that time :)

**Reply by nagyrobi** [reply]
Yes but you publish firmware binary version for ESP32-S3 too, I assume there exists PCB project for it...

**Reply by dzurikmiroslav** [reply]
Not yet, I start building S3 binaries after issue https://github.com/dzurikmiroslav/esp32-evse/issues/24
I think somebody could make minimal schematic with S3 on breadboard, so at least the OTA will work for him.

**Comment by nagyrobi** [comment]
I see now that there are two binaries for ESP32-S2, `esp32s2-evse.bin` and `esp32s2da.bin`. Which one is for the PCB you linked above (esp32s2-evse-d-a)?

**Reply by dzurikmiroslav** [reply]
Files esp32-evse.bin, esp32s2-evse.bin, esp32s3-evse.bin are only application partition, this file is used for upgrade firmware or OTA.

Files esp32s2da.bin, esp32devkitc.bin are merged partitions (bootloader, application, configuration, ota_data...) with dedicated board config, this is used for first flash, next time you cant update only app partition or do OTA.

Official documentation about partitions https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/partition-tables.html

**Comment by Radwan-Alalili** [comment]
Hi they,

I hope you doing well,

I have a question about the esp32 which you use in the first module (single phase) which is ESP32-WROOM-32D or as you said in the readme (ESP32-DEVKITC-32D), I could not find this one in my area is it ok to use ESP-WROOM-32.

And I have other question if you didn't mind, does the output of the both (single phase & three phase) is the same or there is a different

**Reply by dzurikmiroslav** [reply]
Hi,
yes you could use directly ESP32-WROOM-32D (is also placet on DEVKITC). Also can use any ESP32 with at least 4MB of flash memory.

Minimal schematic (without energy meter) is same with 1/3 phases.

**Comment by nagyrobi** [comment]
Tesla Wall Connector Gen2 box from a defective unit seems too perfect to be repurposed as ESP32-EVSE:
<img width="886" height="657" alt="image" src="https://github.com/user-attachments/assets/929f367d-c043-4b03-a12d-0c3b9d5be943" />
Even ideal from the ESP32 antenna perspective.
Just needs a 3D-printed holder inside the case.
